classes/Twig/Block/UpgradeButtonBlock.php

```diff
@@ -121,8 +121,18 @@ public function render()
 
         if (!in_array($channel, array('archive', 'directory')) && !empty($this->upgrader->version_num)) {
             $latestVersion = "{$this->upgrader->version_name} - ({$this->upgrader->version_num})";
+            $phpCompatibleVersions = $this->selfCheck->phpCompatibleVersions();
+            if (version_compare(substr(PHP_VERSION,0,3), $phpCompatibleVersions[1], '>=') && version_compare(substr(PHP_VERSION,0,3), $phpCompatibleVersions[1], '<=')) {
```

Comment:
You should use substr twice, maybe use variable for a better understanding.
And attribute it to the $phpIsCOmpatible variable directory instead of useless if / else